base/commands/cluster/cluster_list_members.go

```diff
@@ -0,0 +1,190 @@
+			},
+			output.Column{
+				Name:  "Public Address",
+				Value: info.PublicAddress,
+				Type:  serialization.TypeString,
+			},
+			output.Column{
+				Name:  "Private Address",
+				Value: info.PrivateAddress,
+				Type:  serialization.TypeString,
+			},
+			output.Column{
+				Name:  "Hazelcast Version",
+				Value: info.Version,
+				Type:  serialization.TypeString,
+			},
+			output.Column{
+				Name:  "IsMaster",
+				Value: info.Master,
+				Type:  serialization.TypeBool,
+			},
+			output.Column{
+				Name:  "IsLite",
+				Value: info.LiteMember,
+				Type:  serialization.TypeBool,
+			},
+		}
+		if ec.Props().GetBool(clc.PropertyVerbose) {
+			row = append(row,
+				output.Column{
+					Name:  "Member State",
+					Value: info.MemberState,
+					Type:  serialization.TypeString,
+				},
+				output.Column{
+					Name:  "Name",
+					Value: info.Name,
+					Type:  serialization.TypeString,
+				},
+			)
+		}
+		rows = append(rows, row)
+	}
+
+	ec.AddOutputRows(ctx, rows...)
+	if err != nil {
+		return err
+	}
+	stop()
+	return nil
+}
+
+type memberData struct {
+	Order          int64
+	PrivateAddress string
+	PublicAddress  string
+	UUID           string
+	Version        string
+	LiteMember     bool
+	MemberState    string
+	Master         bool
+	Name           string
+}
+
+func newMemberData(m cluster.MemberInfo, order int64) *memberData {
+	priv, pub := findMemberAddresses(m)
+	return &memberData{
+		Order:          order,
+		PrivateAddress: priv,
+		PublicAddress:  pub,
+		UUID:           m.UUID.String(),
+		Version:        fmt.Sprintf("%d.%d.%d", m.Version.Major, m.Version.Minor, m.Version.Patch),
+		LiteMember:     m.LiteMember,
+	}
+}
+
+func findMemberAddresses(m cluster.MemberInfo) (string, string) {
+	pub := ""
+	priv := ""
+	for key, val := range m.AddressMap {
+		if key.Type != cluster.EndpointQualifierTypeClient {
+			continue
+		}
+		switch key.Identifier {
+		case "":
+			priv = val.String()
+		case "public":
+			pub = val.String()
+		}
+	}
+	return priv, pub
+}
+func (m *memberData) enrichMemberData(s control.TimedMemberState) {
+	m.Master = s.Master
+	m.MemberState = s.MemberState.NodeState.State
+	m.Name = s.MemberState.Name
+}
+
+func GetMemberInfos(ctx context.Context, ci *hazelcast.ClientInternal) (map[types.UUID]*memberData, error) {
+	activeMemberList := ci.OrderedMembers()
+	activeMembers := make(map[types.UUID]*memberData, len(activeMemberList))
+	for i, memberInfo := range activeMemberList {
+		state, err := fetchTimedMemberState(ctx, ci, memberInfo.UUID)
+		if err != nil {
+			return nil, err
+		}
+		this := memberInfo.Address
+		returned := state.TimedMemberState.MemberState.Address
+		if string(this) != string(returned) {
+			return nil, fmt.Errorf("Timed member state returned info for wrong member, this: %s, returned: %s", this, returned)
+		}
+		activeMembers[memberInfo.UUID] = newMemberData(memberInfo, int64(i))
+		activeMembers[memberInfo.UUID].enrichMemberData(state.TimedMemberState)
```

Comment:
not a major thing, but you could eliminate `enrichMemberData` func, and pass `state.TimedMemberState` to `newMemberData`.

Reply:
Done 👍🏼